name.abuchen.portfolio/src/name/abuchen/portfolio/model/InvestmentPlan.java

```diff
@@ -417,8 +417,8 @@ else if (planType == AccountTransaction.class)
                             converter.with(targetCurrencyCode).getRate(tDate, security.getCurrencyCode()).getValue());
         }
 
-        long shares = Math
-                        .round(availableAmount * Values.Share.factor() * Values.Quote.factorToMoney() / (double) price);
+        long shares = Math.round((double) availableAmount * Values.Share.factor() * Values.Quote.factorToMoney()
+                        / (double) price);
```

Comment:
The second cast actually is redundant now.

Reply:
Looks like it is still good enough with ```double```. At other places, I have converted calculations with shares and prices to ```BigDecimal``` because both have 8 digits precision now.

Thanks for fixing.